src/providers/tikzcodelense.ts

```diff
@@ -0,0 +1,70 @@
+import * as vscode from 'vscode'
+
+export class TikzCodeLense implements vscode.CodeLensProvider {
+    onDidChangeCodeLenses?: vscode.Event<void> | undefined
+
+    async provideCodeLenses(document: vscode.TextDocument) : Promise<vscode.CodeLens[]> {
+        const matches = findTikzPictures(document)
+        return matches.map(
+            match =>
+                new vscode.CodeLens(match.range, {
+                    title: 'View TikzPicture',
+                    tooltip: 'Open view of this TikzPicture',
+                    command: 'latex-workshop.viewtikzpicture',
+                    arguments: [match.document, match.range]
+                })
+        )
+    }
+}
+
+interface TikzPictureMatch {
+    document: vscode.TextDocument
+    range: vscode.Range
+}
+
+function findTikzPictures(document: vscode.TextDocument) {
+    const matches: TikzPictureMatch[] = []
+    const startRegex = /\\begin{tikzpicture}/
```

Comment:
Use /^\s*\\begin{tikzpicture}/ to prevent the regex from maching a commented-out one.

Reply:
That will introduce issues with getting a useful match index straight up. I can just preprocess the line with `stripComments` from `utils`. This is done in 5a26f26746f0419d3252bd770a7f5fd8956e5f3e